### PR TITLE
CEDS-5998 Enable users to amend declarations even in case of DMSRES notifications

### PR DESCRIPTION
**Pull request by udinobi**

### CODE REVIEW

File: app/models/declaration/submissions/Submission.scala [hunk 1]
```diff
@@ -41,12 +41,12 @@ case class Submission(
     } yield notificationSummaries.map(_.enhancedStatus)
   ).getOrElse(Seq.empty[EnhancedStatus])
 
-  lazy val blockAmendments: Boolean = latestDecId.isEmpty
+  lazy val blockOldAmendments: Boolean = latestDecId.isEmpty
```
Comment (glidester) [inline app/models/declaration/submissions/Submission.scala:44]: Just thinking the name `blockOldAmendments` might be confusing to our successors. Maybe leave as `blockAmendments` and add a comment explaining this `isEmpty` will only be true for legacy declarations created before this change?

File: app/models/declaration/submissions/SubmissionStatus.scala [hunk 2]
```diff
@@ -25,16 +25,14 @@ object SubmissionStatus extends Enumeration {
   implicit val format: Format[SubmissionStatus.Value] = EnumJson.format(SubmissionStatus)
 
   val PENDING, REQUESTED_CANCELLATION, ACCEPTED, RECEIVED, REJECTED, UNDERGOING_PHYSICAL_CHECK, ADDITIONAL_DOCUMENTS_REQUIRED, AMENDED, RELEASED,
-    CLEARED, CANCELLED, CUSTOMS_POSITION_GRANTED, CUSTOMS_POSITION_DENIED, GOODS_HAVE_EXITED_THE_COMMUNITY, DECLARATION_HANDLED_EXTERNALLY,
```
Comment (glidester) [inline app/models/declaration/submissions/SubmissionStatus.scala:28]: Why have you removed `GOODS_HAVE_EXITED_THE_COMMUNITY` sub status?

Comment (udinobi) [reply app/models/declaration/submissions/SubmissionStatus.scala:28]: Never  used

Comment (glidester) [reply app/models/declaration/submissions/SubmissionStatus.scala:28]: Yes please. It is certainly used on the BE https://github.com/hmrc/customs-declare-exports/blob/ad2daf7d76c82ad14de2f44ecdbb1c8e2036803e/app/uk/gov/hmrc/exports/models/declaration/submissions/SubmissionStatus.scala#L47. Need to keep FE and BE in sync

**Review by glidester — APPROVED** [review]

**Pull request merged**